apps/aecore/lib/aecore/naming/tx/name_claim_tx.ex

```diff
@@ -126,10 +126,10 @@ defmodule Aecore.Naming.Tx.NameClaimTx do
           tx_type_state
         ) :: :ok | {:error, String.t()}
   def preprocess_check(tx, sender, account_state, fee, _nonce, _block_height, naming_state) do
-    pre_claim_commitment = Naming.create_commitment_hash(tx.name, tx.name_salt)
+    {:ok, pre_claim_commitment} = Naming.create_commitment_hash(tx.name, tx.name_salt)
```

Comment:
this will fail if `tx.name` is not valid, which is the first occurrence this will happen in a naming workflow, should be handled more nice (best in the `cond do`). This was the point were I stopped in my try to refactor this.

Reply:
Well, as far as I see it, if the `tx.name` is invalid, the `validate()` func will fail, so `preprocess_check()` won't even be called